### PR TITLE
CM-31147 - Add support reading existing restore file from directory for gradle

### PR DESCRIPTION
**Pull request by morsa4406**

### CODE REVIEW

File: cycode/cli/files_collector/sca/maven/base_restore_maven_dependencies.py [hunk 1]
```diff
@@ -39,6 +39,22 @@ def get_manifest_file_path(self, document: Document) -> str:
...
+            with open(restore_file_path) as file:
+                restore_file_content = file.read()
+        else:
+            restore_file_content = execute_command(self.get_command(manifest_file_path), manifest_file_path,
+                                                   self.command_timeout)
+
+        return Document(restore_file_path, restore_file_content, self.is_git_diff)
+
+    def _verify_restore_file_already_exist(self, restore_file_path: str) -> bool:
+        return False
```
Comment (MarshalX) [inline cycode/cli/files_collector/sca/maven/base_restore_maven_dependencies.py:56]: imo we should mark this as an abstract method like others instead without any implementation (`pass`)

File: cycode/cli/files_collector/sca/maven/base_restore_maven_dependencies.py [hunk 2]
```diff
@@ -39,6 +39,22 @@ def get_manifest_file_path(self, document: Document) -> str:
             else document.path
         )
 
+    def try_restore_dependencies(self, document: Document) -> Optional[Document]:
+        manifest_file_path = self.get_manifest_file_path(document)
+        restore_file_path = build_dep_tree_path(document.path, self.get_lock_file_name())
+
+        if self._verify_restore_file_already_exist(restore_file_path):
+            with open(restore_file_path) as file:
```
Comment (MarshalX) [inline cycode/cli/files_collector/sca/maven/base_restore_maven_dependencies.py:47]: pls use `get_file_content` instead from `path_utils.py`

File: cycode/cli/files_collector/sca/maven/base_restore_maven_dependencies.py (hunk 1 above)
Comment (MarshalX) [inline cycode/cli/files_collector/sca/maven/base_restore_maven_dependencies.py:56]:
```suggestion
    @abstractmethod
    def _verify_restore_file_already_exist(self, restore_file_path: str) -> bool:
        pass
```

**Review by MarshalX — COMMENTED** [review]
hi pls update your branch and run the code formatted to fix CI. You need to update installed deps after updating the branch

`poetry install` and then `poetry run ruff format .`

**Comment by MarshalX** [comment]
@morsa4406 how can we rename this PR to be in the future changelog of CLI? I mean to be understanding for others what was changed

File: cycode/cli/files_collector/sca/maven/base_restore_maven_dependencies.py (hunk 1 above)
Comment (morsa4406) [reply cycode/cli/files_collector/sca/maven/base_restore_maven_dependencies.py]: Fixed

File: cycode/cli/files_collector/sca/maven/restore_maven_dependencies.py [hunk 3]
```diff
@@ -29,6 +29,9 @@ def get_command(self, manifest_file_path: str) -> List[str]:
     def get_lock_file_name(self) -> str:
         return join_paths('target', MAVEN_CYCLONE_DEP_TREE_FILE_NAME)
 
+    def verify_restore_file_already_exist(self, restore_file_path: str) -> bool:
+        False
```
Comment (MarshalX) [inline cycode/cli/files_collector/sca/maven/restore_maven_dependencies.py:33]: looks like you missed "return" word

Comment (morsa4406) [reply cycode/cli/files_collector/sca/maven/restore_maven_dependencies.py]: Fixed

**Review by MarshalX — APPROVED** [review]

**Pull request merged**